open-company/open-company-web
# Post link behaviour

Card: https://trello.com/c/HgO3wJBw

Related PRs:
- [Storage](https://github.com/open-company/open-company-storage/pull/179)
- [Auth](https://github.com/open-company/open-company-auth/pull/71)
- [Lib](https://github.com/open-company/open-company-lib/pull/48)
- [Interaction](https://github.com/open-company/open-company-interaction/pull/27)
- [Digest](https://github.com/open-company/open-company-digest/pull/12)
- [Email](https://github.com/open-company/open-company-email/pull/36)
- [Bot](https://github.com/open-company/open-company-bot/pull/48)

Possible post outcome: https://docs.google.com/document/d/1gTWQHQsPSKT6gMg5kMZIGm_gd5trsPJJKh0HrWkhMl8/edit

To test:
- check that all the possible scenarios are respected (also that mobile version is ok)
- make sure the secure post is loaded w/o comments and reactions if no id-token is provided
- make sure the secure post view loads comments and reactions when visited with a valid id-token
- make sure you can react and comment if you have an id-token
- make sure the JWT take precedence over the id-token if you visit a secure post url with an id-token with a logged in user
- make sure 
- remove alpha from lib
- release lib non alpha version
- change lib version to all
- increase lib version to next alpha
- commit and merge

### Added Email and Bot PRs:
I had to add 2 new PRs, since i changed the id-token format to have a key called `:teams` instead of `:team-id` i had to update those 2 repo too after [this Nathan's work](https://github.com/open-company/open-company-notify/pull/5). I know the id-token will always have only one team since we use that for a single page but storage and auth read that like they read the JWT so it's better to keep that in sync (IMHO). Ping me with questions
